File: Selenium/Project/Activity7.py
from selenium import webdriver
from selenium.webdriver.common.action_chains import ActionChains
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
from selenium.webdriver.common.by import By
with webdriver.Firefox() as driver:
    driver.get("http://alchemy.hguy.co/crm")

    userName = driver.find_element(By.ID,"user_name")
    userName.send_keys("admin")
    
    password = driver.find_element(By.ID,"username_password")
    password.send_keys("pa$$w0rd")
	
    login = driver.find_element(By.ID,"bigbutton")
    login.click()

    salesButton = driver.find_element(By.XPATH,"//*[@id=\"grouptab_0\"]")
    actions = ActionChains(driver)
    actions.move_to_element(salesButton).perform()

    wait = WebDriverWait(driver, 10)
    leadOption = wait.until(EC.visibility_of_element_located((By.XPATH, "//*[@id=\"moduleTab_9_Leads\"]")))
    leadOption.click()

    wait = WebDriverWait(driver, 30)
    additionalInfo = wait.until(EC.visibility_of_element_located((By.CSS_SELECTOR, "[title='Additional Details']")))

    additionalInfo.click()

    phoneNumber = driver.find_element(By.CSS_SELECTOR,".phone")
    print(f"The phonenumber displayed on the pop up of leads is {phoneNumber.text} ")

    driver.quit()

File: Selenium/Project/Activity7.java
package seleniumActivities;

import java.time.Duration;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.testng.annotations.AfterClass;
import org.testng.annotations.BeforeClass;
import org.testng.annotations.Test;

public class Activity7 {
	WebDriver driver;
	
	@BeforeClass
	public void setUp() {
		driver = new FirefoxDriver();
		driver.get("https://alchemy.hguy.co/crm/");
		WebElement userName = driver.findElement(By.id("user_name"));
		WebElement password = driver.findElement(By.id("username_password"));
		WebElement login = driver.findElement(By.id("bigbutton"));
		
		userName.sendKeys("admin");
		password.sendKeys("pa$$w0rd");
		login.click();
	}
	
	@Test
	public void readaPopUp() {
		WebElement salesButton = driver.findElement(By.xpath("//*[@id=\"grouptab_0\"]"));
		Actions actions = new Actions(driver);
        actions.moveToElement(salesButton).perform();
        
        WebDriverWait wait1 = new WebDriverWait(driver, Duration.ofSeconds(30));
        WebElement leadOption = wait1.until(ExpectedConditions.visibilityOfElementLocated(
            By.xpath("//*[@id=\"moduleTab_9_Leads\"]")));
        
        leadOption.click();
        WebDriverWait wait2 = new WebDriverWait(driver, Duration.ofSeconds(10));
        WebElement additionalInfo = wait2.until(ExpectedConditions.visibilityOfElementLocated(
        		By.cssSelector("[title='Additional Details']")));
        
        additionalInfo.click();
        
        WebElement phoneNumber = driver.findElement(By.cssSelector(".phone"));
        System.out.println("The phone number displayed on the leads pop up is: "+phoneNumber.getText());   
		
	}
	
	@AfterClass
	public void tearDown() {
		driver.quit();
	}

}
